tools/compare_pages.sh

```diff
@@ -64,22 +64,52 @@ file_mod_time() {
   fi
 }
 
+# Check if cached PDF is up-to-date
+is_pdf_current() {
+  local pdf_file="$1"
+
+  if [[ ! -f "$pdf_file" ]]; then
+    return 1
+  fi
+
+  # Check if poppler is available
+  if command -v pdftotext >/dev/null 2>&1; then
+    # Get 7-character commit SHA from main branch (matching PDF format)
+    local pattern
+    pattern=$(git --no-pager log main -1 --format="%h" 2>/dev/null)
```

Comment:
I can easily have my local version of main branch outdated, then this will claim that the file is always up to date

Reply:
Maybe that's a pretty niche case, but it's annoying to download every hour. I changed to firstly check the time, and then compare it with remote SHA, would that work?

Reply:
And/or increase it to 3 hours? 1h is too short.